packages/mendel-middleware/index.js

```diff
@@ -29,6 +30,38 @@ function MendelMiddleware(opts) {
             return getPath({ bundle: bundle, hash: tree.hash });
         };
 
+        req.mendel.getExposed = function(variations) {
+            if (!variations) {
+                variations = 'base';
+            }
+
+            if (exposedBundleVariations[variations]) {
```

Comment:
There is a number of reasons you can't cache like this:

1. This variations are an array, so each request is a new reference, so this is a memory leak.
2. `['base', 'bucket_A']` is the same as `['bucket_A', 'base']` and the logic to normalize, sanitize and order by configuration precedence is inside `trees` so you the proper place to cache would be there.
3. Even if you sanitize, sort and join the variations, this is a vector for a DDOS attack because you can create infinite keys (although the objects would be the same).
4. In multi tenant deployments (one process per processor, between 4 and 16 processors per server and many servers per datacenter, like yahoo has) you will store cache in each process and consume a lot of memory.

I think these are all the reasons we don't cache in Mendel middleware or in `MendelTrees`